src/sql/src/plan/statement/dml.rs

```diff
@@ -160,25 +166,128 @@ pub fn plan_read_then_write(
     }))
 }
 
+fn is_side_effecting(
+    scx: &StatementContext,
+    select: &SelectStatement<Aug>,
+) -> Result<Option<SideEffectingFunction>, PlanError> {
+    match &select {
+        SelectStatement {
+            query:
+                Query {
+                    ctes: CteBlock::Simple(ctes),
+                    body: SetExpr::Select(body),
+                    order_by,
+                    limit: None,
+                    offset: None,
+                },
+            as_of: None,
+        } if ctes.is_empty() && order_by.is_empty() => match std::ops::Deref::deref(body) {
```

Comment:
I think you just want the `*` operator here? But I didn't actually try this.

```suggestion
        } if ctes.is_empty() && order_by.is_empty() => match *body {
```